dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceFileHelper.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.dockstore.webservice.helpers;
+
+import io.dockstore.common.DescriptorLanguage.FileTypeCategory;
+import io.dockstore.webservice.core.SourceFile;
+import io.dockstore.webservice.core.WorkflowVersion;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.LoaderOptions;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+
+public final class SourceFileHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceFileHelper.class);
+
+    private SourceFileHelper() {
+
+    }
+
+    /**
+     * Finds the primary descriptor in a workflow version.
+     * @param workflowVersion
+     * @return
+     */
+    public static Optional<SourceFile> findPrimaryDescriptor(WorkflowVersion workflowVersion) {
+        return workflowVersion.getSourceFiles().stream()
+            .filter(sf -> Objects.equals(sf.getPath(), workflowVersion.getWorkflowPath()))
+            .findFirst();
+    }
+
+    /**
+     * Finds all test files in a workflow version
+     * @param workflowVersion
+     * @return
+     */
+    public static List<SourceFile> findTestFiles(WorkflowVersion workflowVersion) {
```

Comment:
Would it make sense to move `findPrimaryDescriptor` and `findTestFiles` into `WorkflowVersion`?

Reply:
Hmm, perhaps

Reply:
I hesitated because our Hibernate entities are mainly just persistent properties with getters/setters, with a few exceptions.

But the code otherwise does seem to make more sense in WorkflowVersion, so I will move.